Makefile.common

```diff
@@ -89,7 +89,7 @@ RUNTIME_HEADERS := $(wildcard $(ROOTDIR)/runtime/caml/*.tbl) \
 REQUIRED_HEADERS := $(RUNTIME_HEADERS) $(wildcard *.h)
 endif
 
-%.$(O): %.c $(RUNTIME_HEADERS) $(wildcard *.h)
+%.$(O): %.c $(REQUIRED_HEADERS) $(wildcard *.h)
```

Comment:
I'm confused, if `$(REQUIRED_HEADERS)` is either empty or `$(RUNTIME_HEADERS) $(wildcard *.h)`, how can this make a difference given that `$(wildcard *.h)` is already a dependency? (Could this be a matter of the wildcard being evaluated at a different time?).

Reply:
No, that's another artefact of the change... too many variables being renamed and refactored in the review, and the semantics get messed up!

Reply:
The point here is that if `COMPUTE_DEPS` is `false` then you want to depend on `$(RUNTIME_HEADERS)` (which is everything in `runtime/caml/`) **and** any headers in the current build directory. If `COMPUTE_DEPS` is not `false`, then `$(DEP_CC)` will have built the `.d` files indicating exactly which `.h` files to depend on.

It's a relatively minor optimisation in this `Makefile`, because in the directories apart from `runtime` which have `.c` files, changing a header file in that directory does usually rebuild everything (`unixsupport.h`, for example)

Reply:
I still don't understand what this commit is or was fixing.

The question applies in fact to both the old state (as you initially submitted the PR) and the new state, after your new commit "Tighten dependencies":
- The old change looks like a no-op to me, so was it really fixing an issue? What issue? (What the change wrong, or am I missing something?)
- With the new change you make in "Tighten dependencies", then we end up with no .h dependencies at all if COMPUTE_DEPS is false. Is that the point of the change to this file, or another beneficial change?

Reply:
The old change in this PR accidentally a no-op, yes - while adding the empty case for `$(RUNTIME_HEADERS)` I noticed that the wrong variable was being used below it, but didn't fully take in why it was all still OK.

The pattern rules are not normally supposed to mention dependencies (cf. the `.cmo` and `.cmx` suffix/pattern rules) - you have separate specific rules which list the dependencies. `$(REQUIRED_HEADERS)` is to allow _development_ of the compiler even with `--disable-dependency-generation` (so `$(COMPUTE_DEPS)` is `false`). In that case, you won't have the specific dependencies for each `.o` file as they're not generated. So the idea is that each `.o` file depends on every single `.h` file - so any change to any header rebuilds all `.o` files (it's the best you can do in the absence of dependency information).

The mistake is that before I'd left one of those cases for `$(COMPUTE_DEPS)` = `false` in global scope.

(the reason for this is that it means that instead of _requiring_ developers on MSVC to have GCC to work on the compiler, we _recommend_ developers on MSVC have GCC because incremental compilation of the compiler will be faster. It also means that there is not a configuration where incremental compilation of the C sources is actually broken)